### ISSUE TITLE
PostgreSQL: Change columns TOAST strategies (PLAIN , EXTERNAL , EXTENDED) in the program interface

### ISSUE THREAD

**Issue by Evgeny-V-K** [dbeaver/dbeaver#35212]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I can't change the storage strategy (PLAIN , EXTERNAL , EXTENDED) of a column in the program interface, and when I can't see then in script.

**Comment by E1izabeth** [comment]
Thank you for request.

We can read this with the [following query](https://stackoverflow.com/a/49947950), show in UI in Properties editor and allow changing (sql for that is `ALTER COLUMN column_name SET STORAGE { PLAIN | EXTERNAL | EXTENDED | MAIN | DEFAULT }`).

![image](https://github.com/user-attachments/assets/f5e19fb0-d473-4cee-8228-8351e65150bf)

**Comment by KawtharAlakri** [comment]
Hi @E1izabeth, I would like to work on this. Can you assign it to me? & if you can guide me in which plugin I will find this feature in the codebase.

**Comment by KawtharAlakri** [comment]
I started working on this issue, but not sure if I'm on the right track, struggle to find where the call to the DB should happen (the query). Can someone please take a look into the changes I made so far in [#35212-postgresql-columns-storage-strategy](https://github.com/KawtharAlakri/dbeaver/tree/%2335212-postgresql-columns-storage-strategy) and give me some feedback/guidance?

**Comment by E1izabeth** [comment]
Hi, @KawtharAlakri
Sorry for long response. As far as I see, you're on the right way.
When you edit some column property, `PostgreTableColumnManager::addObjectModifyActions` is called.

**Comment by gagan-bhullar-tech** [comment]
@E1izabeth can you please assign this to me

**Comment by E1izabeth** [comment]
@gagan-bhullar-tech we assign issues only to our team members. It's enough if you state in the comment that you're working on it

**Comment by gagan-bhullar-tech** [comment]
@E1izabeth thanks. Will work on this.

**Comment by eusebe-cda** [comment]
Hi, I'll try it too if you agree...

**Comment by kseniaguzeeva** [comment]
Unfortunately unnecessary data appears in table DDL in PostgreSQL 14. It doesn't happen in 16, 17 PostgreSQLs.
```
CREATE TABLE test_review."35212" (
	id int4 NOT NULL,
	testplain varchar NULL,
	testmain varchar NULL,
	testexternal varchar NULL,
	testextended varchar NULL,
	CONSTRAINT "35212_pk" PRIMARY KEY (id)
);

-- Column storage

ALTER TABLE test_review."35212" ALTER COLUMN tableoid SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN cmax SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN xmax SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN cmin SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN xmin SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN ctid SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN id SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN testplain SET STORAGE PLAIN;
ALTER TABLE test_review."35212" ALTER COLUMN testmain SET STORAGE MAIN;
ALTER TABLE test_review."35212" ALTER COLUMN testexternal SET STORAGE EXTERNAL;
ALTER TABLE test_review."35212" ALTER COLUMN testextended SET STORAGE EXTENDED;
```

![Image](https://github.com/user-attachments/assets/48591341-2f6d-4ee0-af79-e4ef5acd62ab)

**Comment by eusebe-cda** [comment]
> Unfortunately unnecessary data appears in table DDL in PostgreSQL 14. It doesn't happen in 16, 17 PostgreSQLs.
> 
> ```
You're right. I fixed it, and also added a check to only modify the generated DDL if the specified storage is different from the default storage of the selected column type...

thanks !

**Comment by kseniaguzeeva** [comment]
We are close to the finish line. Few found issues to fix:
1. All tested PostgreSQL (10,12, 14, 16, 17): when a storage type of an existing column is changed to empty and saved after that, the value returns to previous one after refresh. It looks buggy.Is it possible to exclude empty value from this dropdown? I remember that we spoke about empty value for new columns, but this case differs a little.

https://github.com/user-attachments/assets/9cb03063-4917-4add-a6c1-87eb2a6d6237


2. PostgreSQL 10,12, 14 don't have default value for storage. Is it expected?

**Comment by eusebe-cda** [comment]
> We are close to the finish line. Few found issues to fix:
> 
> 1. All tested PostgreSQL (10,12, 14, 16, 17): when a storage type of an existing column is changed to empty and saved after that, the value returns to previous one after refresh. It looks buggy.Is it possible to exclude empty value from this dropdown? I remember that we spoke about empty value for new columns, but this case differs a little.
> 
>  1t.mp4 
> 2. PostgreSQL 10,12, 14 don't have default value for storage. Is it expected?

1. I haven't found how to prevent selecting an empty value in the column listView. The data type can also be set to empty, but in this case it generates an exception. I'm not sure I can do better...
2. Yes, this is intentional: old versions of postgresql don't support the DEFAULT keyword

**Comment by kseniaguzeeva** [comment]
1. Ok, we can keep the current implementation
2. thank you
3. One more issue has been found: the setting has appeared in Postgresql forks. E.g. it exists in Cockroach, but fails when a non-default storage is selected for a column. Could you please add the new property `org.jkiss.dbeaver.ext.postgresql.model.PostgreServerExtension`?

**Comment by eusebe-cda** [comment]
Thanks for the tests. I'll try with PostgreServerExtension. The goal is to completely disable the feature in forks, correct (not even showing the storage of an attribute) ?

**Comment by E1izabeth** [comment]
Right

**Comment by eusebe-cda** [comment]
I think it should be ok with the latest changes, but I couldn't test with Cockroach...

**Comment by kseniaguzeeva** [comment]
Thank you for your pull request. Verified (PostgreSQL, Cockroach, Greenplum have been tested)
